File: app/src/main/java/com/dmko/pairwisecomparison/ui/screens/comparisons/ComparisonsContract.java
package com.dmko.pairwisecomparison.ui.screens.comparisons;

import com.dmko.pairwisecomparison.data.entities.Comparison;
import com.dmko.pairwisecomparison.ui.base.mvp.BasePresenter;
import com.dmko.pairwisecomparison.ui.base.mvp.BaseView;

import java.util.List;

public interface ComparisonsContract {

    interface View extends BaseView {

        void showLoading(boolean isLoading);

        void setComparisons(List<Comparison> comparisons);

        void setEmptyComparisons();

        void showAddEditComparisonDialog(String comparisonId);

        void openComparison(Comparison comparison);
    }

    interface Presenter extends BasePresenter<View> {

        void start();

        void openComparisonSelected(Comparison comparison);

        void addComparisonSelected();

        void updateComparisonSelected(Comparison comparison);

        void deleteComparisonSelected(Comparison comparison);
    }
}

File: app/src/main/java/com/dmko/pairwisecomparison/ui/screens/comparisons/ComparisonsContract.kt
package com.dmko.pairwisecomparison.ui.screens.comparisons

import com.dmko.pairwisecomparison.data.entities.Comparison
import com.dmko.pairwisecomparison.ui.base.mvp.BasePresenter
import com.dmko.pairwisecomparison.ui.base.mvp.BaseView

interface ComparisonsContract {

    interface View : BaseView {

        fun showLoading(isLoading: Boolean)

        fun setComparisons(comparisons: List<Comparison>)

        fun setEmptyComparisons()

        fun showAddEditComparisonDialog(comparisonId: String?)

        fun openComparison(comparison: Comparison)

        fun openCopyComparisonDialog(comparisonId: String)
    }

    interface Presenter : BasePresenter<View> {

        fun start()

        fun openComparisonSelected(comparison: Comparison)

        fun addComparisonSelected()

        fun updateComparisonSelected(comparison: Comparison)

        fun deleteComparisonSelected(comparison: Comparison)

        fun copyComparisonSelected(comparison: Comparison)
    }
}